.github/workflows/archive_build.yml

```
@@ -1,20 +1,29 @@
+      - name: Fetch binary
+        env:
+          firmwareUrl : https://github.com/jpcornil-git/ayabAsync-firmware/releases/download/${{ env.AYAB_VERSION }}/firmware_uno_1_1_1.hex
+        run: |
+          set -e
+          if ! wget --spider ${firmwareUrl}; then
+            echo "Error: Firmware file not found"
+            exit 1
+          fi
+          wget ${firmwareUrl}
+      - name: Verify hex file presence
+        run: |
```

actionlint

shellcheck reported issue in this script: SC2144:error:1:11: -f doesn't work with globs. Use a for loop

(shellcheck)